contracts/CollectiveGovernance.sol

```diff
@@ -100,27 +111,27 @@ contract CollectiveGovernance is Governance, VoteStrategy, ERC165 {
     }
 
     modifier requireVoteReady(uint256 _proposalId) {
-        require(_storage.isFinal(_proposalId), "Voting is not ready");
+        if(!(_storage.isFinal(_proposalId)) revert CG__VotingNotReady();
         _;
     }
 
     modifier requireVoteClosed(uint256 _proposalId) {
-        require(_storage.isFinal(_proposalId) && !isVoteOpenByProposalId[_proposalId], "Vote is not closed");
+        if(_storage.isFinal(_proposalId) && isVoteOpenByProposalId[_proposalId]) revert CG__VotingOpen();
         _;
     }
 
     modifier requireVoteOpen(uint256 _proposalId) {
-        require(_storage.isFinal(_proposalId) && isVoteOpenByProposalId[_proposalId], "Voting is closed");
+        if(!(_storage.isFinal(_proposalId) && isVoteOpenByProposalId[_proposalId])) revert CG__VotingClosed();
         _;
     }
 
     modifier requireVoteAllowed(uint256 _proposalId) {
-        require(!_storage.isCancel(_proposalId) && !_storage.isVeto(_proposalId), "Vote cancelled");
+        if(_storage.isCancel(_proposalId) && _storage.isVeto(_proposalId) revert CG__VotingCancelled();
         _;
     }
 
     modifier requireSupervisor(uint256 _proposalId) {
-        require(_storage.isSupervisor(_proposalId, msg.sender), "Supervisor required");
+        if(!(_storage.isSupervisor(_proposalId, msg.sender)) revert CG__SupervisorRequired();
```

Comment:
extra parens here

Reply:
see below similar issues...please double check and fix all cases and make sure there is a test to prove correctness.